js/packages/mobile-wallet-adapter-protocol-web3js/src/transact.ts

```diff
@@ -1,258 +1,281 @@
+                                    payloads,
+                                });
+                                const signatures = base64EncodedSignatures.map(toUint8Array).map(bs58.encode);
+                                return signatures as TransactionSignature[];
+                            } as Web3RemoteMobileWallet[TMethodName];
+                            break;
+                        case 'signMessages':
+                            target[p] = async function ({
+                                payloads,
+                                ...rest
+                            }: Parameters<Web3MobileWallet['signMessages']>[0]) {
+                                const base64EncodedPayloads = payloads.map(fromUint8Array);
+                                const { signed_payloads: base64EncodedSignedMessages } = await wallet.signMessages({
+                                    ...rest,
+                                    payloads: base64EncodedPayloads,
+                                });
+                                const signedMessages = base64EncodedSignedMessages.map(toUint8Array);
+                                return signedMessages;
+                            } as Web3RemoteMobileWallet[TMethodName];
+                            break;
+                        case 'signTransactions':
+                            target[p] = async function ({
+                                transactions,
+                                ...rest
+                            }: Parameters<Web3MobileWallet['signTransactions']>[0]) {
+                                const payloads = transactions.map(getPayloadFromTransaction);
+                                const { signed_payloads: base64EncodedCompiledTransactions } =
+                                    await wallet.signTransactions({
+                                        ...rest,
+                                        payloads,
+                                    });
+                                const compiledTransactions = base64EncodedCompiledTransactions.map(toUint8Array);
+                                const signedTransactions = compiledTransactions.map(getTransactionFromWireMessage);
+                                return signedTransactions;
+                            } as Web3RemoteMobileWallet[TMethodName];
+                            break;
+                        default: {
+                            target[p] = wallet[p] as unknown as Web3RemoteMobileWallet[TMethodName];
+                            break;
+                        }
+                    }
+                }
+                return target[p];
+            },
+            defineProperty() {
+                return false;
+            },
+            deleteProperty() {
+                return false;
+            },
+        });
+        return callback(augmentedAPI);
+    };
+    return await baseTransactRemote(augmentedCallback, config);
+}
+
+export async function startRemoteScenario(

```

Comment:
weird diff. This method was added to augment the base `startRemoteScenario` method. The `augmentWalletAPI` below was also added to remove code duplication between `startRemoteScenario` and `transact`